File: C++/624. Maximum Distance in Arrays.cpp
class Solution {
public:
    int maxDistance(vector<vector<int>>& a) {
        int m,M,i,x=0;
        m=a[0][0];
        M=a[0][a[0].size()-1];
        for(i=1;i<a.size();i++){
            x=max(x,max(abs(m-a[i][a[i].size()-1]),abs(M-a[i][0])));
            m=min(m,a[i][0]);
            M=max(M,a[i][a[i].size()-1]);
        }
        return x;
    }
};

File: JAVA/624. Maximum Distance in Arrays.java
import java.util.List;

class Solution {
    public int maxDistance(List<List<Integer>> a) {
        int m, M, i, x = 0;
        m = a.get(0).get(0);
        M = a.get(0).get(a.get(0).size() - 1);
        for (i = 1; i < a.size(); i++) {
            x = Math.max(x, Math.max(Math.abs(m - a.get(i).get(a.get(i).size() - 1)),Math.abs(M - a.get(i).get(0))));
            m = Math.min(m, a.get(i).get(0));
            M = Math.max(M, a.get(i).get(a.get(i).size() - 1));
        }
        return x;
    }
}